numpy/lib/function_base.py

```diff
@@ -4014,122 +4116,232 @@ def _quantile_is_valid(q):
     return True
 
 
-def _lerp(a, b, t, out=None):
-    """ Linearly interpolate from a to b by a factor of t """
-    diff_b_a = subtract(b, a)
-    # asanyarray is a stop-gap until gh-13105
-    lerp_interpolation = asanyarray(add(a, diff_b_a*t, out=out))
-    subtract(b, diff_b_a * (1 - t), out=lerp_interpolation, where=t>=0.5)
-    if lerp_interpolation.ndim == 0 and out is None:
-        lerp_interpolation = lerp_interpolation[()]  # unpack 0d arrays
+def virtual_index_formula(
+        array_size: np.array, quantiles: np.array, alpha: float, beta: float
+) -> np.array:
+    """
+   Compute the floating point indexes of an array for the linear
+   interpolation of quantiles.
+
+   Note:
+       Compared to R, -1 is added because R array start at 1 (0 for python)
+   """
+    return array_size * quantiles + (
+            alpha + quantiles * (1 - alpha - beta)
+    ) - 1
+
+
+def gamma_formula(val: np.array, val_floor: np.array) -> np.array:
+    """
+    Compute the gamma (a.k.a 'm' or weight) for the linear interpolation
+    of quantiles.
+    For the extreme values 0 and 1,
+     when gamma is 0 the left bound will be picked,
+     when gamma is 1 the right bound will be.
+    """
+    return val - val_floor
```

Comment:
This is a bizarre way to spell subtraction. Why not just use subtraction?

Reply:
In the most simple cases, gamma is just x - floor(x) but depending on the interpolation method it may be different.
I will refactor this into to have a `::get_gamma` which will handle all the gamma cases in one place.